search/includes/classes/class-versioning.php

```diff
@@ -531,7 +531,11 @@ public function get_next_version_number( $versions ) {
 		$new_version = null;
 
 		if ( ! empty( $versions ) && is_array( $versions ) ) {
-			$new_version = max( array_keys( $versions ) );
+			$versions = array_map( function ( $v ) {
+				return is_int( $v ) ? $v : 0;
+			}, array_keys( $versions ) );
+
+			$new_version = max( $versions );
```

Comment:
[`max()`](https://www.php.net/max) works differently in PHP 8:

> **Caution**  Be careful when passing arguments of different types because **max()** can produce unpredictable results.